backend/controller/dal/dal.go

```diff
@@ -159,12 +158,18 @@ type Reservation interface {
 	Rollback(ctx context.Context) error
 }
 
+type artefactDAL struct {
```

Comment:
I don't think you need this intermediate DAL layer. Just split the queries out of the controller's sqlc.yaml into a separate destination directory (see other examples), and use the generated sqlc queries directly in `artefacts.DALRegistry`

Reply:
thanks for the pointer, applied